cleanup/aws_snapshots.sh

```diff
@@ -47,10 +47,25 @@ snapshot_ids="$(aws ec2 describe-snapshots \
   --region=us-east-2 \
   | jq -r '.[][]')"
 
-if [ -n "${snapshot_ids}" ]
-then
-  export -f run_aws_ec2_command
-  echo "${snapshot_ids}" | parallel --halt-on-error never --no-run-if-empty run_aws_ec2_command delete-snapshot --snapshot-id {} :::
+if [ -n "${snapshot_ids}" ]; then
+  echo "== Found the following snapshots for deletion:"
+  echo "${snapshot_ids}"
+  echo "======"
+
+  for snapshot_id in ${snapshot_ids}; do
+    echo "== Attempting to delete snapshot: ${snapshot_id}"
+    if ! output=$(run_aws_ec2_command delete-snapshot --snapshot-id "${snapshot_id}" 2>&1); then
```

Comment:
This is not the direction we want to have. Checking "after tentative deletion" is an unwanted pattern:
- It creates a lot of unneeded requests
- It slows down the whole process

=> what you want is to check whether a snapshot is used or not.

Some resources to help you:

- https://stackoverflow.com/questions/29499216/how-to-check-whether-a-snapshot-with-id-is-in-use-of-aws-amis-in-boto
- https://docs.aws.amazon.com/ebs/latest/userguide/ebs-describing-snapshots.html
- https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-snapshots.html